qualtran/bloqs/qsp_bloq_test.py

```diff
@@ -0,0 +1,24 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import numpy as np
+
+from qualtran.bloqs.qsp_bloq import qsp_phase_factors
+
+
```

Comment:
Can we add a test that checks for unitary correctness of the final `QSP` block? I'm thinking of a parameterized test that takes in:
1) `U` : A random single / two qubit unitary generated using `cirq.testing.random_unitary`. You can create a test class that derives from `GateWithRegisters` and has this unitary.  
2) `P`, `Q`: Coefficients (both real / complex?) for polynomials with degree 2, 3 & 4

and then assert that top-left and bottom-left blocks of `cirq.unitary(QSP(U, P, Q))` correspond to `P(U)` `Q(U)` respectively?

Reply:
Yep will add this. Unfortunately there are no examples in the paper for computing Q from P, so I'll hand-craft some polynomials for now.